master/internal/logretention/logretention.go

```diff
@@ -0,0 +1,115 @@
+package logretention
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-co-op/gocron/v2"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const retainForever = -1
+
+var (
+	log = logrus.WithField("component", "log-retention")
+
+	schedulerDefaultOpts = []gocron.SchedulerOption{
+		gocron.WithLimitConcurrentJobs(1, gocron.LimitModeReschedule),
+	}
+	scheduler gocron.Scheduler
+
+	// WaitGroup is used for testing purposes to wait for the log retention scheduler to finish.
+	WaitGroup *sync.WaitGroup
```

Comment:
If this is used for testing only and it is exported, I think the var should be named something ridiculous like `TestingOnlySynchronizationHelper`, ha.